exercises/change/example.go

```diff
@@ -0,0 +1,119 @@
+// Package change demonstrates making change with fewest number of coins.
+package change
+
+// Change returns a list of coins in increasing value order to make
+// change for target amount from the coins given; an infinite supply of coins is assumed.
+func Change(coins []int, target int) (result []int) {
+	solutions := make([][]int, 0)
+	if target == 0 {
+		result = []int{}
+	} else {
+		if target > 0 {
+			findSolutions(coins, target, &solutions)
+		}
+		if len(solutions) == 0 {
+			result = []int{-1}
+		} else if len(solutions) == 1 {
+			result = solutions[0]
+		} else {
+			result = chooseBestSolution(solutions)
+		}
+	}
```

Comment:
I've heard [Frances Campoy](https://www.youtube.com/channel/UC_BzFbxG2za3bp5NRRRXJSw) say an else block can almost be considered a code smell. Personally I'd return from the function early than re-assigning to the result var.

```
	if target == 0 {
		return []int{}
	}
	if target > 0 {
		findSolutions(coins, target, &solutions)
	}
	if len(solutions) == 0 {
		return []int{-1}
	}
	if len(solutions) == 1 {
		return solutions[0]
	}
	return chooseBestSolution(solutions)
```